optuna/integration/tensorboard.py

```diff
@@ -0,0 +1,106 @@
+import optuna
+from optuna import type_checking
+import os
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+    from typing import Optional  # NOQA
+
+try:
+    import tensorboard
+    from tensorboard.plugins.hparams import api as hp
+    import tensorflow as tf
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    hp = object
+    tensorboard = object
+    tensorflow = object
+
+
+def _check_tensorboard_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            "TensorBoard is not available. Please install TensorBoard to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "tensorboard`. For further information, please refer to the installation guide "
+            "of TensorBoard. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
+class TensorBoardCallback(object):
+    """Callback to track Optuna trials with TensorBoard.
+
+    This callback adds relevant information that is tracked by Optuna to TensorBoard.
+    Args:
+        metric_name:
+            Name of the metric. Since the metric itself is just a number,
+            `metric_name` can be used to give it a name. So you know later
+            if it was roc-auc or accuracy.
```

Comment:
Please specify whole information of all arguments. (`dirname ` and `param_distributions`)